core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -183,6 +183,15 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
     case TRIM:
       unparseTrim(writer, call, leftPrec, rightPrec);
       break;
+    case ITEM:
```

Comment:
`ITEM` is not only used for `array`, but also for `map`, `struct`. So it's better to check whether the `operand(0)` is an `array`.
Besides, does `ORDINAL` only work in the array accessing in BigQuery? Are there any other places that also need to be handled like this?

Reply:
`ORIDANL` is an [Array subscript operator](https://cloud.google.com/bigquery/docs/reference/standard-sql/operators#array_subscript_operator) in BigQuery. I didn't see any other places needs this operator.

BTW, because `arrray subscript operator` has `ORDIANL`, `SAFE_ORDINAL`, `OFFST` and `SAFE_OFFSET`, should we offer a config for the user to switch which one is preferred?